### ISSUE TITLE
Confirmation prompt before exiting to Home -- Avoid closing OsmAnd inadvertently

### ISSUE THREAD

**Issue by Max1234-Ita** [osmandapp/OsmAnd#20689]
### Describe the idea

Not sure if this has already been proposed, I tried searching but found nothing.

While driving, I keep my smartphone on my car's dashboard; wpedaling on my bike or just hiking I usually hold the same with one hand just to take a quick look to the map: in those cases, I need to interact with OsmAnd with a single hand.

Sometimes it happens that when I'm browsing the map or try to unzoom o recenter to current position twice, I inadvertently tap the _Back_ button and OsmAnd exits immediately to the Home: after this I need to tap the app icon again, and wait for it to load, initialize, etc.: quite fristrating, actually.

A prompt should be displayed to the user, asking to confirm that he/she really wants to go back to the Home screen. This should happen either the Navigation is active or OsmAnd is just in "Browse map" mode.

### Expected behaviour

- When hitting the Back button of the device, a textbox should appear, with some text like "Hit Back twice to exit" or, alternatively, a small modal box with a confirmation button and a similar message, i.e. "Please confirm you really want to exit" + "Exit" button (or so); A short countdown (say, 3 seconds) should also start at the same time.

- If the _Back_ or the chosen confirmation button is pressed, then OsmAnd will be closed/put into background and the Home screen appear, as it currently happens (no change in this behaviour);

- If a tap occurs in any location different from the _Back_ or confirmation button, the prompt should disappear and OsmAnd will stay actve and visible;

- The same as above should happen if the 3-second countdown expires: the prompt disappears and OsmAnd continues working.

This would avoid the user to inadvertently close the app, i.e. just because he/she's holding the device with a single hand and try to just scroll the map a little, center on current position or to decrease the zoom with the ( - ) button (in some devices, the Back button is on the right, very close to them, see attached screenshot).

![image](https://github.com/user-attachments/assets/c9ada4f3-ce31-43e3-b9b8-0211c56a778f)

### Alternatives you've considered

Being very careful while handling the device is a good practice but not always comfortable, not the safer: it also requires additional attention, resulting in a potential distraction from other activity, such as driving (sometimes the user just needs to interact with the map to have just a quick look to the surroundings or the next turn).

The prompt to exit would make the app more usable and safer, as the user would not be distracted while trying to restart it.

### Context

Use cases: 
 - Driving, biking, hiking and all those cases when the user might be able to hold the device with just one hand;
 - Users with poor vision from the close distance (as myself, when driving and wearing glasses with long-distance lenses :-) ) might also be helped by this feature.

**Comment by vshcherb** [comment]
Yes it was proposed as an idea with another task to implement "Proper exit" (kill app once back is pressed 2 times)

**Comment by SpitFire-666** [comment]
I too keep hitting Back (force of habit) during navigation, expecting it to ask if I want to cancel navigation - instead it exits the app
